pkg/auth/event_policy.go

```diff
@@ -0,0 +1,147 @@
+
+					relevantPolicies = append(relevantPolicies, policy)
+					break // no need to check the other .spec.to's from this policy
+				}
+			}
+
+			if to.Selector != nil {
+				parts := strings.Split(to.Selector.APIVersion, "/")
+				if len(parts) != 2 {
+					return nil, fmt.Errorf("cannot split apiVersion into group and version: %s", to.Selector.APIVersion)
+				}
+
+				if strings.EqualFold(parts[0], resourceGVK.Group) &&
+					strings.EqualFold(to.Selector.Kind, resourceGVK.Kind) {
+
+					selector, err := metav1.LabelSelectorAsSelector(to.Selector.LabelSelector)
+					if err != nil {
+						return nil, fmt.Errorf("failed to parse selector: %w", err)
+					}
+
+					if selector.Matches(labels.Set(resourceObjectMeta.Labels)) {
+						relevantPolicies = append(relevantPolicies, policy)
+						break // no need to check the other .spec.to's from this policy
+					}
+				}
+			}
+		}
+	}
+
+	return relevantPolicies, nil
+}
+
+// ResolveSubjects returns the OIDC service accounts names for the objects referenced in the EventPolicySpecFrom.
+func ResolveSubjects(resolver *resolver.AuthenticatableResolver, eventPolicy *v1alpha1.EventPolicy) ([]string, error) {
+	allSAs := []string{}
+	for _, from := range eventPolicy.Spec.From {
+		if from.Ref != nil {
+			sas, err := resolveSubjectsFromReference(resolver, *from.Ref, eventPolicy)
+			if err != nil {
+				return nil, fmt.Errorf("could not resolve subjects from reference: %w", err)
+			}
+			allSAs = append(allSAs, sas...)
+		} else if from.Sub != nil {
+			allSAs = append(allSAs, *from.Sub)
+		}
+	}
+
+	return allSAs, nil
+}
+
+func resolveSubjectsFromReference(resolver *resolver.AuthenticatableResolver, reference v1alpha1.EventPolicyFromReference, trackingEventPolicy *v1alpha1.EventPolicy) ([]string, error) {
+	authStatus, err := resolver.AuthStatusFromObjectReference(&corev1.ObjectReference{
+		APIVersion: reference.APIVersion,
+		Kind:       reference.Kind,
+		Namespace:  reference.Namespace,
+		Name:       reference.Name,
+	}, trackingEventPolicy)
+
+	if err != nil {
+		return nil, fmt.Errorf("could not resolve auth status: %w", err)
+	}
+
+	objSAs := authStatus.ServiceAccountNames
+	if authStatus.ServiceAccountName != nil {
+		objSAs = append(objSAs, *authStatus.ServiceAccountName)
+	}
+
+	objFullSANames := make([]string, 0, len(objSAs))
+	for _, sa := range objSAs {
+		objFullSANames = append(objFullSANames, fmt.Sprintf("system:serviceaccount:%s:%s", reference.Namespace, sa))
+	}
+
+	return objFullSANames, nil
+}
+
+// SubjectContained checks if the given sub is contained in the list of allowedSubs
+// or if it matches a prefix pattern in subs (e.g. system:serviceaccounts:my-ns:*)
+func SubjectContained(sub string, allowedSubs []string) bool {
+	for _, s := range allowedSubs {
+		if strings.EqualFold(s, sub) {
+			return true
+		}
+
+		if strings.HasSuffix(s, "*") &&
+			strings.HasPrefix(strings.ToLower(sub), strings.TrimSuffix(s, "*")) {
+			return true
```

Comment:
why `ToLower` ?

Reply:
Thought initially that this can be some user input. Will update it